lldb/source/Plugins/TypeSystem/Swift/TypeSystemSwiftTypeRef.cpp

```diff
@@ -2991,6 +2997,12 @@ TypeSystemSwiftTypeRef::GetCanonicalType(opaque_compiler_type_t type) {
     ConstString mangled(mangling.result());
     return GetTypeFromMangledTypename(mangled);
   };
+  auto impl = [&] {
+    auto mangled_name = StringRef(AsMangledName(type));
+    return m_canonical_types_cache.GetOrCreate(
```

Comment:
I don't think this is necessarily correct, since constructing the canonical demangle tree involves DWARF lookups, but they are not part of the cache key here.

Reply:
In what cases might a DWARF lookup return different results? When we add a new module to the module list?

Reply:
Yes that would be one such case.

Reply:
Generally, a good way to be sure that we understand all implicit dependencies would be to make the function we want to cache astatic function first, then all dependencies become explicit, and we know they need to go into the cache key (or invalidate the cache).